tests/utils/test_cache.py

```diff
@@ -19,3 +19,37 @@ def test_cached_object(self):
         with self.assertRaises(Expired) as e:
             obj.unwrap(after)
         self.assertEqual(e.exception.args, (expiry,))
+
+
+class ExpiringCacheTestCase(TestCase):
+
+    def test_expiring_cache(self):
+        expiry = Timestamp('2014')
+        before = expiry - Timedelta('1 minute')
+        after = expiry + Timedelta('1 minute')
+
+        cache = ExpiringCache()
+
+        cache.set('foo', 1, expiry)
+        cache.set('bar', 2, after)
+
+        self.assertEqual(cache.get('foo', before), 1)
+        # Unwrap on expiry is allowed.
+        self.assertEqual(cache.get('foo', expiry), 1)
+
+        with self.assertRaises(KeyError) as e:
+            self.assertEqual(cache.get('foo', after))
+            self.assertEqual(e.exception.args, ('foo',))
+
+        # Should raise same KeyError after deletion.
+        with self.assertRaises(KeyError) as e:
+            self.assertEqual(cache.get('foo', after))
+            self.assertEqual(e.exception.args, ('foo',))
+
+        # Second value should still exist.
+        self.assertEqual(cache.get('bar', after), 2)
```

Comment:
This reads a little misleadingly.  I'd just give `bar` a different expiration entirely.